src/Core/Components/Icons/Icon.cs

```diff
@@ -87,15 +87,28 @@ internal Icon InverseColor(bool accentContainer)
     /// </summary>
     public virtual MarkupString ToMarkup(string? size = null, string? color = null)
     {
-        var styleWidth = size ?? $"{(int)Size}px";
+        var styleWidth = size ?? $"{Width}px"; // was previously using (int)Size which is what Width does. Width is also the property used in FluentIcon.StyleValue
         var styleColor = color ?? Color ?? "var(--accent-fill-rest)";
         return new MarkupString($"<svg viewBox=\"0 0 {(int)Size} {(int)Size}\" fill=\"{styleColor}\" style=\"background-color: var(--neutral-layer-1); width: {styleWidth};\" aria-hidden=\"true\">{Content}</svg>");
     }
 
+    internal virtual int? OverriddenWidth { get; private set; } = null; // init to null so the previous behavior is unchanged
+
+    /// <summary>
+    /// Sets the render size of the icon in px. Use null to revert to icon original size
+    /// </summary>
+    /// <param name="size"></param>
+    /// <returns></returns>
+    public virtual Icon WithSize(int? size)
```

Comment:
I think it would be possible to have a `string` argument: `WithSize(string ? size)` as in the **ToMarkup** method. This could be interesting for using `%` or other unit types. What do you think?

Reply:
Certainly, it would minimally require changes in the FluentIcon component and that exact ToMarkup method where "px" is hardcoded. It might also require changes in the way Width and Size are delt with (mostly Width which is a protected in an unsealed class). And that's where I decided to play it safe due to my limited knowledge of the ripple effects. 

That said, if you're ok with pushing this a little further, I can spend some more time on this.
I would also be a bit more comfortable if we had some tests coverage here. Looks like we'll have to synch our changes to move forward😊

Reply:
I added a PR #1136 to include Unit Tests

Reply:
I think you meant #1140

Reply:
Why not have both `WithSize(int? size)` and `WithSize(string? size)`?

And why are these parameters nullable? What would be the outcome of sending in a null value? I think we may expect people to supply a valid input value.

Reply:
We could have both, with `WithSize(int? size) => WithSize(size is null ? null : $"{size}px");` 
The advantage of having a null value is to "reset" another specified size.